libc/docs/dev/header_generation.rst

```diff
@@ -1,120 +1,160 @@
+yaml file depending on the header spec or enter it through the command line.
 
-``begin``
-~~~~~~~~~
+To enter through the command line:
 
-This is not a replacement command. It is an error to list it in the input
-``.h.def`` file. It is normally listed in the files included by the
-``include_file`` command (the ``.h.inc`` files). A common use of this command it
-mark the beginning of what is to be included. This prevents copying items like
-license headers into the generated header file.
+1. Make sure you are in the llvm-project directory.
 
-Arguments
+2. Enter in the command line:
+   ``python3 libc/newhdrgen/yaml_to_classes.py
+   libc/newhdrgen/yaml/[yaml_file.yaml] --add_function “<return_type>”
+   <function_name> “<function_arg1, function_arg2>” <standard> <guard>
+   <attribute>``
 
-  None.
+   Example:
+   ``python3 libc/newhdrgen/yaml_to_classes.py
+   libc/newhdrgen/yaml/ctype.yaml --add_function “char” example_funtion “int,
+   void, const void” stdc example_float example_attribute``
+   
+   Keep in mind only the return_type and arguments have quotes around them. If
+   you do not have any guards or attributes you may enter “null” for both.
 
-Action
+3. Check the yaml file that the added function is present. You will also get a
+   generated header file with the new addition in the newhdrgen directory to
+   examine.
 
-  The header generator will only include content starting from the line after the
-  line on which this command is listed.
 
-``public_api``
-~~~~~~~~~~~~~~
+Testing
+-----------------------------
 
-This is a replacement command which should be listed in an input ``.h.def``
-file. The header file generator will replace this command with the public API of
-the target platform. See the build system document for more information on the
-relevant build rules. Also, see "Mechanics of public_api" to learn the mechanics
-of how the header generator replaces this command with the public API.
+New Headergen has an integration test that you may run once you have configured
+your CMake within the build directory. In the command line, enter the following:
+“ninja check -newhdrgen”. The integration test is one test that ensures the
+process of yaml to classes to generate headers works properly. If there are any
+new additions on formatting headers, make sure the test is updated with the
+specific addition.
 
-Arguments
+Integration Test can be found in: ``libc/newhdrgen/tests/test_integration.py``
 
-  None.
+File to modify if adding something to formatting:
+``libc/newhdrgen/tests/expected_output/test_header.h``
 
-Action
 
-  The header generator will replace this command with the public API to be exposed
-  from the generated header file.
+Common Errors
+-----------------------------
+1. ``"/llvm-project/libc/newhdrgen/yaml_to_classes.py", line 67, in yaml_to_classes function_data["return_type"]``
+
+    If you receive this error or any error pertaining to
```

Comment:
The formatting seems a bit odd to me. This might work better if the errors are subheadings and the paragraphs are just normal.